src/os/bluestore/BlueStore.cc

```diff
@@ -9501,9 +9501,19 @@ void BlueStore::_kv_sync_thread()
 	}
       }
 
-      // submit synct synchronously (block and wait for it to commit)
-      int r = cct->_conf->bluestore_debug_omit_kv_commit ? 0 : db->submit_transaction_sync(synct);
-      ceph_assert(r == 0);
+      RocksDBStore::RocksWBHandler bat_txc;
+      if (bluefs_do_check_balance) {
+        RocksDBStore::RocksDBTransactionImpl * _t =
+          static_cast<RocksDBStore::RocksDBTransactionImpl *>(synct.get());
+        _t->bat.Iterate(&bat_txc);
+      }
+      if (!(bluefs_do_check_balance && bat_txc.seen.empty())) {
+        // submit synct synchronously (block and wait for it to commit)
```

Comment:
@wuhongsong Have you considered using WriteBatch::Count() for testing if transaction is empty?

Reply:
@aclamk  Thank you  for your advice, I have modified it.